src/main/java/it/gov/pagopa/pu/workflow/wf/receiptclassification/iuf/classification/IufReceiptClassificationWFImpl.java

```diff
@@ -0,0 +1,122 @@
+
+import io.temporal.spring.boot.WorkflowImpl;
+import it.gov.pagopa.payhub.activities.activity.classifications.*;
+import it.gov.pagopa.payhub.activities.dto.classifications.IufClassificationActivityResult;
+import it.gov.pagopa.pu.workflow.wf.receiptclassification.iuf.dto.IufReceiptClassificationForReportingSignalDTO;
+import it.gov.pagopa.pu.workflow.wf.receiptclassification.iuf.dto.IufReceiptClassificationForTreasurySignalDTO;
+import it.gov.pagopa.pu.workflow.wf.receiptclassification.iuf.config.IufReceiptClassificationWfConfig;
+import it.gov.pagopa.pu.workflow.wf.receiptclassification.iuf.helper.TransferClassificationWfHelperActivity;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+import static it.gov.pagopa.pu.workflow.wf.ingestionflow.paymentsreporting.wfingestion.PaymentsReportingIngestionWFImpl.TASK_QUEUE;
+
+/**
+ * Implementation of the IufReceiptClassificationWF interface
+ */
+@Slf4j
+@WorkflowImpl(taskQueues = TASK_QUEUE)
+public class IufReceiptClassificationWFImpl implements IufReceiptClassificationWF, ApplicationContextAware {
+  public static final String TASK_QUEUE = "IufReceiptClassificationWF";
+
+  private IufReceiptClassificationForTreasurySignalDTO signalDTO;
+
+  private ClearClassifyIufActivity clearClassifyIufActivity;
+  private IufClassificationActivity iufClassificationActivity;
+  private TransferClassificationActivity transferClassificationActivity;
+
+  private TransferClassificationWfHelperActivity transferClassificationWfHelperActivity;
+
+  private IufClassificationActivityResult result;
+
+  /**
+   * Temporal workflow will not allow to use injection in order to avoid
+   * <a href="https://docs.temporal.io/workflows#non-deterministic-change">non-deterministic changes</a> due to dynamic reconfiguration.<BR />
+   * Anyway it allows to override ActivityOptions, but actually it's not supporting the override based on the particular workflow.<BR />
+   * In {@link it.gov.pagopa.pu.workflow.config.TemporalWFImplementationCustomizer} we are already setting defaults to all workflows.<BR />
+   * Use this as an example to override based on the particular workflow.
+   */
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    IufReceiptClassificationWfConfig wfConfig = applicationContext.getBean(IufReceiptClassificationWfConfig.class);
+
+    clearClassifyIufActivity = wfConfig.buildClearClassifyIufActivityStub();
+    iufClassificationActivity = wfConfig.buildIufClassificationActivityStub();
+    transferClassificationActivity = wfConfig.buildTransferClassificationActivityStub();
+
+    transferClassificationWfHelperActivity = wfConfig.buildTransferClassificationStarterHelperActivityStub();
+
+  }
+
+  @Override
+  public void classify() {
+
+    if (result != null && result.isSuccess()) {
```

Comment:
we should flat the result list in order to retrieve the Set of all transfer to classify (a Set in order to remove duplicates)